packages/plugin-react/src/index.ts

```diff
@@ -203,7 +203,14 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
         ])
       }
 
-      if (opts.jsxRuntime === 'classic' && isJSX) {
+      if (
+        opts.jsxRuntime === 'classic' &&
+        isJSX &&
+        // OXC injects self and source so these plugins are not needed for rolldown-vite
```

Comment:
Why is that? Does OXC run before Babel with rolldown-vite?
Technically it works without in default Vite but the source info is off which makes info in React devtools wrong (and breaks https://github.com/ArnaudBarre/vite-plugin-react-click-to-component)

Reply:
Oxc runs after this plugin and injects source and self. Oxc rejects an JSX with self and source with `Duplicate __self prop found.` error.

Reply:
Ah, probably we should set `development: false`.

Reply:
Hmm, it seems it still errors with `development: false`.

Reply:
Made an issue: https://github.com/oxc-project/oxc/issues/10391

Reply:
Ok I see. Yeah we can merge this as this is just a small DX issue